cmd/kubeadm/app/cmd/alpha/kubelet.go

```diff
@@ -109,16 +111,39 @@ func newCmdKubeletConfigDownload() *cobra.Command {
 			client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
 			kubeadmutil.CheckErr(err)
 
-			err = kubeletphase.DownloadConfig(client, kubeletVersion, constants.KubeletRunDirectory)
+			criSocket, err := getCRISocket(criSocketStr)
+			if err != nil {
+				kubeadmutil.CheckErr(err)
+			}
+
+			nodeReg := kubeadmapi.NodeRegistrationOptions{
+				CRISocket: criSocket,
+			}
+
+			err = kubeletphase.DownloadConfig(client, kubeletVersion, &nodeReg, constants.KubeletRunDirectory)
 			kubeadmutil.CheckErr(err)
 		},
 	}
 
 	options.AddKubeConfigFlag(cmd.Flags(), &kubeConfigFile)
+	cmdutil.AddCRISocketFlag(cmd.Flags(), &criSocketStr)
 	cmd.Flags().StringVar(&kubeletVersionStr, "kubelet-version", kubeletVersionStr, "The desired version for the kubelet. Defaults to being autodetected from 'kubelet --version'.")
 	return cmd
 }
 
+func getCRISocket(criSocketStr string) (string, error) {
+	if len(criSocketStr) > 0 {
+		return criSocketStr, nil
+	}
+
+	criSocket, err := kubeadmruntime.DetectCRISocket()
+	if err != nil {
```

Comment:
it can be simplified to `return kubeadmruntime.DetectCRISocket()`
In overall, I don't think we need that function if it is called only in one place. It is easier to inline this if/else in the `newCmdKubeletConfigDownload`

Reply:
This is used to allow customize the CRI Socket when downloading kubelet config file.

The getCRISocket function is very like the getKubeletVersion function, it is used to make code logic simple and easy to read.

Reply:
in any way, construct 

```go
	criSocket, err := kubeadmruntime.DetectCRISocket()
	if err != nil {
		return "", err
	}
	return criSocket, nil
```

can be reduced to 
```go
	return kubeadmruntime.DetectCRISocket()
```